backend/portal/api/portal_api.py

```diff
@@ -731,6 +731,13 @@ def get_dataset_identifiers(url: str):
     if collection is None:  # orphaned datasets - shouldn't happen, but we should return 404 just in case
         raise NotFoundHTTPException()
 
+    # If we look up by canonical_id (i.e. the if does not hold), we can run into issues if the collection
```

Comment:
this edge case can also occur anywhere else we try to look up a pre-publish dataset with its canonical ID. We should either: 1) (longer-term solution) update persistence layer to not map the canonical dataset to a version until publish, and return None (or most recent version) for pre-publish dataset canonical ID calls. 
or 2) (shorter-term solution) add this quick fix somewhere in the flow of `get_dataset_version_from_canonical`

Reply:
will follow up in https://github.com/chanzuckerberg/single-cell-data-portal/pull/4229